lib/puma/thread_pool.rb

```diff
@@ -24,6 +24,44 @@ class ThreadPool
     class ForceShutdown < RuntimeError
     end
 
+    class Worker
```

Comment:
Nit: we already call processes `workers` in too many contexts, I find the word overloaded. we're using it to refer to a thread execution context here. Doesn't Sidekiq call these `Processor`? Just anything other than Worker.

Reply:
Well, `@workers = []` was there before this PR, populated with `Thread`s, so I can see how it was natural with `Worker`. :) Is `WorkerThread` a acceptable name?

Reply:
but maybe that should have been `@threads = []`. Another suggestion: `DeferrableThread`

Reply:
If I'm reading the PR right, all threads in the threadpool are getting a Worker, so `deferrablethread` doesn't make sense to me because _all_ threads are deferrable. This class also represents a kind of an abstract thread, something which lightly wraps the idea of a thread and context but isn't a Thread itself.

I think I just prefer Processor. Mike's really good at naming and there's a reason why he picked that for the ~similar concept in Sidekiq. Will accept other suggestions though.